src/libaktualizr/storage/fsstorage.cc

```diff
@@ -149,102 +149,98 @@ bool FSStorage::loadTlsCert(std::string* cert) { return loadTlsCommon(cert, conf
+    metadata->director_targets = Utils::readFile(director_path / ("targets.json" + suffix));
+    metadata->image_targets = Utils::readFile(image_path / ("targets.json" + suffix));
+    metadata->image_timestamp = Utils::readFile(image_path / ("timestamp.json" + suffix));
+    metadata->image_snapshot = Utils::readFile(image_path / ("snapshot.json" + suffix));
   }
 
-  json = Utils::parseJSONFile(image_path / "root.json");
-  if (json != Json::nullValue) {
-    Json::Value fixed_json = json;
-    if (!json.isMember("signed")) {
-      fixed_json["signed"] = json;
-    }
-    metadata->image_root = Uptane::Root("image", fixed_json);
-  } else {
-    return false;
-  }
+  return true;
+}
 
-  json = Utils::parseJSONFile(director_path / "targets.json");
-  if (json != Json::nullValue) {
-    Json::Value fixed_json = json;
-    if (!json.isMember("signed")) {
-      fixed_json["signed"] = json;
-    }
-    metadata->director_targets = Uptane::Targets(fixed_json);
-  }
+void FSStorage::clearMetadataCommon(const std::string& suffix) {
+  boost::filesystem::path image_path = Utils::absolutePath(config_.path, config_.uptane_metadata_path) / "repo";
+  boost::filesystem::path director_path = Utils::absolutePath(config_.path, config_.uptane_metadata_path) / "director";
 
-  json = Utils::parseJSONFile(image_path / "targets.json");
-  if (json != Json::nullValue) {
-    Json::Value fixed_json = json;
-    if (!json.isMember("signed")) {
-      fixed_json["signed"] = json;
-    }
-    metadata->image_targets = Uptane::Targets(fixed_json);
-  }
+  boost::filesystem::remove(director_path / ("root.json" + suffix));
+  boost::filesystem::remove(director_path / ("targets.json" + suffix));
+  boost::filesystem::remove(director_path / ("root.json" + suffix));
+  boost::filesystem::remove(director_path / ("targets.json" + suffix));
+  boost::filesystem::remove(image_path / ("timestamp.json" + suffix));
+  boost::filesystem::remove(image_path / ("snapshot.json" + suffix));
+}
 
-  json = Utils::parseJSONFile(image_path / "timestamp.json");
-  if (json != Json::nullValue) {
-    Json::Value fixed_json = json;
-    if (!json.isMember("signed")) {
-      fixed_json["signed"] = json;
-    }
-    metadata->image_timestamp = Uptane::TimestampMeta(fixed_json);
-  }
+void FSStorage::storeMetadata(const Uptane::RawMetaPack& metadata) { storeMetadataCommon(metadata, ""); }
+bool FSStorage::loadMetadata(Uptane::RawMetaPack* metadata) { return loadMetadataCommon(metadata, ""); }
+void FSStorage::clearMetadata() { clearMetadataCommon(""); }
 
-  json = Utils::parseJSONFile(image_path / "snapshot.json");
-  if (json != Json::nullValue) {
-    Json::Value fixed_json = json;
-    if (!json.isMember("signed")) {
-      fixed_json["signed"] = json;
-    }
-    metadata->image_snapshot = Uptane::Snapshot(fixed_json);
-  }
+void FSStorage::storeUncheckedMetadata(const Uptane::RawMetaPack& metadata) { storeMetadataCommon(metadata, ".raw"); }
```

Comment:
Ok, I think I get what the suffix is. Metadata should only be stored in the exact form in which it was read from the wire. All processing has the possibility to break signatures (most notably, parsing it to a `struct` that doesn't capture a field the backend sends).